packages/desktop-client/src/components/select/DateSelect.tsx

```diff
@@ -341,9 +372,9 @@ export default function DateSelect({
             } else {
               setValue(selectedValue || '');
 
-              let date = d.parse(selectedValue, dateFormat, new Date());
-              if (date instanceof Date && !isNaN(date)) {
-                onSelect(d.format(date, 'yyyy-MM-dd'));
+              let date = parse(selectedValue, dateFormat, new Date());
+              if (date instanceof Date && !isNaN(date.valueOf())) {
```

Comment:
Typescript was complaining about ```isNaN(date)``` because date is typed as a Date (previous check ensures that) - isNaN only accepts a number.

under the hood isNaN would convert the date object to a timestamp number - to satisfy typescript we now have to call that manually.